eclair-core/src/test/scala/fr/acinq/eclair/router/RouteCalculationSpec.scala

```diff
@@ -56,6 +56,30 @@ class RouteCalculationSpec extends FunSuite {
     assert(route.map(hops2Ids) === Success(1 :: 2 :: 3 :: 4 :: Nil))
   }
 
+  test("check fee against max pct properly") {
+
+    // fee is acceptable is it is either
+    // - below our maximum fee base
+    // - below our maximum fraction of the paid amount
+
+    // here we here a maximum fee base of 1 msat, and all our updates have a base fee of 10 msat
+    // so our fee will always be above the base fee, and we will always check that is is below our maximum percentage
+    // of the amount being paid
+
+    val updates = List(
+      makeUpdate(1L, a, b, MilliSatoshi(10), 10, cltvDelta = 1),
+      makeUpdate(2L, b, c, MilliSatoshi(10), 10, cltvDelta = 1),
+      makeUpdate(3L, c, d, MilliSatoshi(10), 10, cltvDelta = 1),
+      makeUpdate(4L, d, e, MilliSatoshi(10), 10, cltvDelta = 1)
+    ).toMap
+
+    val g = makeGraph(updates)
+
+    val route = Router.findRoute(g, a, e, DEFAULT_AMOUNT_MSAT, numRoutes = 1, routeParams = DEFAULT_ROUTE_PARAMS.copy(maxFeeBase = MilliSatoshi(1)))
```

Comment:
I don't see any test verifying that:
* Fee percent are ok but fee base is not -> no route found
* Fee base are ok but fee percent is not -> no route found

Do we have that somewhere? Or would it be useful to add?

Reply:
It's because the actual test is fee base below threshold OR fee percent below threshold so in both cases we should find a route. The bug was that if fee base was above threshold but fee percent was acceptable we would still fail to find a route.